src/common/tests_dmg_helpers.c

```diff
@@ -582,7 +582,15 @@ parse_device_info(struct json_object *smd_dev, device_list *devices,
 
 	for (i = 0; i < dev_length; i++) {
 		dev = json_object_array_get_idx(smd_dev, i);
-		strcpy(devices[*disks].host, strtok(host, ":") + 1);
+
+		if (strlen(host + 2) <= DSS_HOSTNAME_MAX_LEN)
```

Comment:
Does (host+2) instead of (host) points to the real host name?

devices[*disks].host is an 50bytes array:
typedef struct {
        uuid_t          device_id;
        char            state[10];
        int             rank;
        char            host[50];
        int             tgtidx[MAX_TEST_TARGETS_PER_DEVICE];
        int             n_tgtidx;
}  device_list;

and 
#define DSS_HOSTNAME_MAX_LEN    255
Won't there still a risk of Copy into fixed size buffer (STRING_OVERFLOW)?

Reply:
host[50] has been changed to host[DSS_HOSTNAME_MAX_LEN]. Please see the changes below..

```
typedef struct {
	uuid_t		device_id;
	char		state[10];
	int		rank;
	char		host[DSS_HOSTNAME_MAX_LEN];
	int		tgtidx[MAX_TEST_TARGETS_PER_DEVICE];
	int		n_tgtidx;
}  device_list;
```